iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/MemoryRequestCheck.java

```diff
@@ -19,13 +19,36 @@
  */
 package org.sonar.iac.kubernetes.checks;
 
+import java.util.Collection;
+import java.util.Set;
+import javax.annotation.Nullable;
 import org.sonar.check.Rule;
+import org.sonar.iac.kubernetes.model.LimitRange;
+import org.sonar.iac.kubernetes.model.LimitRangeItem;
 
 @Rule(key = "S6873")
 public class MemoryRequestCheck extends AbstractRequestCheck {
   private static final String MESSAGE = "Specify a memory request for this container.";
   private static final String KEY = "memory";
 
+  private static final Set<String> LIMIT_TYPES = Set.of("Pod", "Container");
+
+  @Override
+  protected boolean hasLimitDefinedGlobally(Collection<LimitRange> globalResources) {
+    return globalResources.stream()
+      .flatMap(limitRange -> limitRange.limits().stream())
+      .anyMatch(MemoryRequestCheck::hasMemoryRequest);
+  }
+
+  private static boolean hasMemoryRequest(LimitRangeItem limitRangeItem) {
+    var defaultMemoryRequest = limitRangeItem.defaultRequestMap().get(KEY);
+    return LIMIT_TYPES.contains(limitRangeItem.type()) && isValidMemory(defaultMemoryRequest);
+  }
+
+  static boolean isValidMemory(@Nullable String value) {
```

Comment:
Should it be extracted to utils and shared between the two rules?

Reply:
Yes, I agree. I wasn't sure where to put it, but have an idea now.